File: src/main/java/me/kuku/yuq/logic/NeTeaseLogic.java
package me.kuku.yuq.logic;

import com.IceCreamQAQ.Yu.annotation.AutoBind;
import me.kuku.pojo.Result;
import me.kuku.yuq.entity.NeTeaseEntity;

import java.io.IOException;

@AutoBind
public interface NeTeaseLogic {
    Result<NeTeaseEntity> loginByPhone(String username, String password) throws IOException;
    String sign(NeTeaseEntity neTeaseEntity) throws IOException;
    String listeningVolume(NeTeaseEntity neTeaseEntity) throws IOException;
}

File: src/main/kotlin/me/kuku/yuq/logic/NeTeaseLogic.kt
package me.kuku.yuq.logic

import com.IceCreamQAQ.Yu.annotation.AutoBind
import me.kuku.yuq.entity.NeTeaseEntity
import me.kuku.yuq.entity.QQLoginEntity
import me.kuku.yuq.pojo.CommonResult

@AutoBind
interface NeTeaseLogic {
    fun loginByPhone(username: String, password: String): CommonResult<NeTeaseEntity>
    fun loginByQQ(qqLoginEntity: QQLoginEntity): CommonResult<NeTeaseEntity>
    fun sign(neTeaseEntity: NeTeaseEntity): String
    fun listeningVolume(neTeaseEntity: NeTeaseEntity): String
}